Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/common/HDIEnvironment.java

```diff
@@ -44,26 +44,25 @@ private HDIEnvironment(Map<String, String> endpoints, Environment environment) {
     }
 
     public HDIEnvironment(@NotNull Environment environment) {
-        switch (environment) {
-            case GLOBAL:
-                this.environment = Environment.GLOBAL;
-                this.endpoints = HDIEnvironment.GLOBAL.endpoints;
-                break;
-            case CHINA:
-                this.environment = Environment.CHINA;
-                this.endpoints = HDIEnvironment.CHINA.endpoints;
-                break;
-            case GERMAN:
-                this.environment = Environment.GERMAN;
-                this.endpoints = HDIEnvironment.GERMANY.endpoints;
-                break;
-            case US_GOVERNMENT:
-                this.environment = Environment.US_GOVERNMENT;
-                this.endpoints = HDIEnvironment.US_GOVERNMENT.endpoints;
-                break;
-            default:
-                this.environment = Environment.GLOBAL;
-                this.endpoints = HDIEnvironment.GLOBAL.endpoints;
+        if (Environment.GLOBAL.equals(environment)) {
```

Comment:
What's the reason to replace `switch case` with `if else`? To improve perf based on frequency of candidates?

Reply:
Since the `Environment` is changed from `Enum` to `Interface`, The values, such as `Environment.GLOBAL` can't be const to be used by `switch-case`, I have to replace them by `if-else`.

Reply:
In fact, I prefer Polymorphism, but current change is minimal.